ygen/genstate.go

```diff
@@ -202,14 +227,14 @@ func (s *genState) buildGoStructDefinitions(entries map[string]*yang.Entry, comp
 
 // findEnumSet walks the list of enumerated value leaves and determines whether
 // code generation is required for each enum. Particularly, it removes
-// duplication between config and state containers when compressOCPaths is true.
+// duplication between config and state containers when compressPaths is true.
 // It also de-dups references to the same identity base, and type definitions.
-func (s *genState) findEnumSet(entries map[string]*yang.Entry, compressOCPaths bool) (map[string]*yangGoEnum, []error) {
+func (s *genState) findEnumSet(entries map[string]*yang.Entry, compressPaths bool) (map[string]*yangGoEnum, []error) {
 	validEnums := make(map[string]*yang.Entry)
 	var enumNames []string
 	var errs []error
 
-	if compressOCPaths {
+	if compressPaths {
```

Comment:
At a high level I wonder if this type of OC specific handling can be extracted out into helpers somehow so that the base code is truly decoupled from the target schema.

Reply:
The problem is that it is quite pervasive throughout the code what we do when compress paths is on, for example, naming is also dependent upon this. So here, we want to search for the enums and calculate their names, if we don't do this here, then we would have to duplicate somewhere else having to do the name calculation. For finding other entities, it's much more complex, since the set of mapped things, as well as their names, is entirely dependent upon the path compression.

Reply:
ack. This was a random thought, just seeing a lot of complexity in the code (including my own) that's path compression related.